--- pcmanfm/settings.cpp	2023-01-04 00:02:47.779200646 +1300
+++ pcmanfm/settings.cpp.patched	2023-01-04 00:04:14.074206555 +1300
@@ -356,12 +356,12 @@
     settings.endGroup();
 
     settings.beginGroup(QStringLiteral("Search"));
-    searchNameCaseInsensitive_ = settings.value(QStringLiteral("searchNameCaseInsensitive"), false).toBool();
-    searchContentCaseInsensitive_ = settings.value(QStringLiteral("searchContentCaseInsensitive"), false).toBool();
-    searchNameRegexp_ = settings.value(QStringLiteral("searchNameRegexp"), true).toBool();
-    searchContentRegexp_ = settings.value(QStringLiteral("searchContentRegexp"), true).toBool();
-    searchRecursive_ = settings.value(QStringLiteral("searchRecursive"), false).toBool();
-    searchhHidden_ = settings.value(QStringLiteral("searchhHidden"), false).toBool();
+    searchNameCaseInsensitive_ = settings.value(QStringLiteral("searchNameCaseInsensitive"), true).toBool();
+    searchContentCaseInsensitive_ = settings.value(QStringLiteral("searchContentCaseInsensitive"), true).toBool();
+    searchNameRegexp_ = settings.value(QStringLiteral("searchNameRegexp"), false).toBool();
+    searchContentRegexp_ = settings.value(QStringLiteral("searchContentRegexp"), false).toBool();
+    searchRecursive_ = settings.value(QStringLiteral("searchRecursive"), true).toBool();
+    searchhHidden_ = settings.value(QStringLiteral("searchhHidden"), true).toBool();
     maxSearchHistory_ = std::clamp(settings.value(QStringLiteral("MaxSearchHistory"), 0).toInt(), 0, 50);
     namePatterns_ = settings.value(QStringLiteral("NamePatterns")).toStringList();
     namePatterns_.removeDuplicates();