--- src/lxdm.c	2022-09-01 08:36:47.000000000 +0200
+++ src/lxdm.c.patched	2024-02-24 23:39:15.319385518 +0100
@@ -44,6 +44,8 @@
 #include <sys/wait.h>
 #include <glib.h>
 #include <glib/gstdio.h>
+#include <utmp.h>
+#include <utmpx.h>
 
 #include <sys/vt.h>
 #include <sys/ioctl.h>
@@ -89,6 +91,8 @@
 	pid_t child;
 	uid_t user;
 	int display;
+	char *username;
+	char *display_full;
 	char *option;	/* hold option in config file */
 	xconn_t dpy;	/* hold this, or X crack */
 	LXDM_AUTH auth;
@@ -347,6 +351,36 @@
 	return s;
 }
 
+static void remove_user_login_from_utmp(const int tty, const char *display)
+{
+    struct utmpx ut;
+    memset(&ut, 0, sizeof(struct utmpx));
+
+    ut.ut_type = DEAD_PROCESS;
+    ut.ut_pid = getpid();
+
+    const char* full_tty;
+    asprintf(&full_tty, "tty%d", tty);
+    strncpy(ut.ut_line, full_tty, sizeof(ut.ut_line) - 1);
+
+    const char* full_display;
+    asprintf(&full_display, "%s.0", display);
+    strncpy(ut.ut_host, full_display, sizeof(ut.ut_host) - 1);
+
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    ut.ut_tv.tv_sec = tv.tv_sec;
+    ut.ut_tv.tv_usec = tv.tv_usec;
+
+    setutxent();
+    pututxline(&ut);
+    endutxent();
+    updwtmpx("/var/log/wtmp", &ut);
+    
+    g_free(full_tty);
+    g_free(full_display);
+}
+
 static void lxsession_stop(LXSession *s)
 {
 	if(s->greeter)
@@ -1168,6 +1202,7 @@
 	int level;
 	LXSession *s=data;
 
+	remove_user_login_from_utmp(s->tty, s->display_full);
 	gchar *argv[] = { "/etc/lxdm/PostLogout", NULL };
 	g_spawn_async(NULL, argv, s->env, G_SPAWN_SEARCH_PATH, NULL, NULL, NULL, NULL);
 
@@ -1351,6 +1386,37 @@
 	g_key_file_free(var);
 }
 
+static void add_user_login_to_utmp(const int tty, const char *display, const char *username)
+{
+    struct utmpx ut;
+    memset(&ut, 0, sizeof(struct utmpx));
+
+    const char* full_tty;
+    asprintf(&full_tty, "tty%d", tty);
+    strncpy(ut.ut_line, full_tty, sizeof(ut.ut_line) - 1);
+    
+    const char* full_display;
+    asprintf(&full_display, "%s.0", display);
+    strncpy(ut.ut_host, full_display, sizeof(ut.ut_host) - 1);    
+
+    strncpy(ut.ut_user, username, sizeof(ut.ut_user) - 1);
+    ut.ut_pid = getpid();
+    ut.ut_type = USER_PROCESS;
+    
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    ut.ut_tv.tv_sec = tv.tv_sec;
+    ut.ut_tv.tv_usec = tv.tv_usec;
+    
+    setutxent();
+    pututxline(&ut);
+    endutxent();
+    updwtmpx("/var/log/wtmp", &ut);
+    
+    g_free(full_tty);
+    g_free(full_display);
+}
+
 void lxdm_do_login(struct passwd *pw, char *session, char *lang, char *option)
 {
 	char *session_name=0,*session_exec=0,*session_desktop_names=0;
@@ -1421,6 +1487,8 @@
 	s->greeter=FALSE;
 	s->idle=FALSE;
 	s->user=pw->pw_uid;
+	s->username=pw->pw_name;
+	s->display_full=getenv("DISPLAY");
 	if(option)
 		s->option=g_strdup(option);
 #if HAVE_LIBCK_CONNECTOR
@@ -1509,7 +1577,9 @@
 		switch_user(pw, session_exec, env);
 		lxdm_quit_self(4);
 	}*/
-	
+
+	add_user_login_to_utmp(s->tty, s->display_full, s->username);
+
 	s->child = pid = lxdm_auth_session_run(&s->auth,session_exec,env);
 	
 	g_free(session_name);
