--- thunar/thunar-shortcuts-model.c	2021-09-18 07:30:41.000000000 +1200
+++ thunar/thunar-shortcuts-model-new.c	2022-01-11 19:34:52.022995822 +1300
@@ -1539,6 +1539,7 @@
                                      ThunarShortcutsModel *model)
 {
   ThunarShortcut *shortcut;
+  GFile          *mount_point;
 
   _thunar_return_if_fail (device_monitor == NULL || THUNAR_DEVICE_MONITOR (device_monitor));
   _thunar_return_if_fail (device_monitor == NULL || model->device_monitor == device_monitor);
@@ -1550,6 +1551,13 @@
   shortcut->device = g_object_ref (device);
   shortcut->hidden = thunar_device_get_hidden (device);
 
+  mount_point = thunar_device_get_root (device);
+  if (mount_point != NULL)
+    {
+      shortcut->file = thunar_file_get (mount_point, NULL);
+      g_object_unref (mount_point);
+    }
+
   switch (thunar_device_get_kind (device))
     {
     case THUNAR_DEVICE_KIND_VOLUME:
@@ -1614,6 +1622,7 @@
   gint            idx;
   GtkTreePath    *path;
   ThunarShortcut *shortcut;
+  GFile          *mount_point;
   gboolean        update_header = FALSE;
 
   _thunar_return_if_fail (THUNAR_DEVICE_MONITOR (device_monitor));
@@ -1637,6 +1646,16 @@
       /* cleanup tooltip */
       g_free (shortcut->tooltip);
       shortcut->tooltip = NULL;
+      
+      if (shortcut->file == NULL)
+        {
+		  mount_point = thunar_device_get_root (device);
+		  if (mount_point != NULL)
+			{
+			   shortcut->file = thunar_file_get (mount_point, NULL);
+			   g_object_unref (mount_point);
+			}
+		}
 
       /* hidden state */
       if (shortcut->hidden != thunar_device_get_hidden (device))
