--- thunar/thunar-file.c	2022-04-03 00:08:34.000000000 +0200
+++ thunar/thunar-file-patched.c	2023-10-19 23:39:26.847039943 +0200
@@ -2442,7 +2442,7 @@
           if (G_LIKELY (info != NULL))
             {
               /* store the new content type */
-              content_type = g_file_info_get_content_type (info);
+              content_type = g_file_info_get_attribute_string (info, G_FILE_ATTRIBUTE_STANDARD_CONTENT_TYPE);
               if (G_UNLIKELY (content_type == NULL))
                 content_type = g_file_info_get_attribute_string (info,
                                                                  G_FILE_ATTRIBUTE_STANDARD_FAST_CONTENT_TYPE);
@@ -2505,7 +2505,7 @@
   if (file->info == NULL)
     return NULL;
 
-  return g_file_info_get_symlink_target (file->info);
+  return g_file_info_get_attribute_byte_string (file->info, G_FILE_ATTRIBUTE_STANDARD_SYMLINK_TARGET);
 }
 
 
@@ -2543,7 +2543,7 @@
   if (file->info == NULL)
     return FALSE;
 
-  return g_file_info_get_is_symlink (file->info);
+  return g_file_info_get_attribute_boolean (file->info, G_FILE_ATTRIBUTE_STANDARD_IS_SYMLINK);
 }
 
 
@@ -2978,8 +2978,8 @@
   if (file->info == NULL)
     return FALSE;
 
-  return g_file_info_get_is_hidden (file->info)
-         || g_file_info_get_is_backup (file->info);
+  return g_file_info_get_attribute_boolean (file->info, G_FILE_ATTRIBUTE_STANDARD_IS_HIDDEN)
+         || g_file_info_get_attribute_boolean (file->info, G_FILE_ATTRIBUTE_STANDARD_IS_BACKUP);
 }
 
 
@@ -3801,7 +3801,7 @@
       if (G_LIKELY (fileinfo != NULL))
         {
           /* take the icon from the info */
-          icon = g_file_info_get_icon (fileinfo);
+          icon = G_ICON (g_file_info_get_attribute_object (fileinfo, G_FILE_ATTRIBUTE_STANDARD_ICON));
           if (G_LIKELY (icon != NULL))
             g_object_ref (icon);
 
