--- src/xfdesktop-special-file-icon.c	2020-04-20 03:34:36.000000000 +1200
+++ src/xfdesktop-special-file-icon-new.c	2022-01-08 10:32:28.732863216 +1300
@@ -484,7 +484,7 @@
             size = g_file_info_get_attribute_uint64(info,
                                                     G_FILE_ATTRIBUTE_STANDARD_SIZE);
 
-            size_string = g_format_size(size);
+            size_string = g_format_size_full(size, G_FORMAT_SIZE_IEC_UNITS);
 
             mtime = g_file_info_get_attribute_uint64(info,
                                                      G_FILE_ATTRIBUTE_TIME_MODIFIED);
