--- src/notification.cpp	2023-09-04 22:38:20.194922617 +0200
+++ src/notification-patched.cpp	2023-09-04 22:39:14.160922102 +0200
@@ -33,7 +33,6 @@
 #include <QDebug>
 #include <XdgIcon>
 #include <KWindowSystem/KWindowSystem>
-#include <KWindowSystem/KX11Extras>
 #include <QMouseEvent>
 #include <QPushButton>
 #include <QStyle>
@@ -327,7 +326,7 @@
         return;
     }
 
-    const auto ids = KX11Extras::stackingOrder();
+    const auto ids = KWindowSystem::stackingOrder();
     for (const WId &i : ids)
     {
         KWindowInfo info = KWindowInfo(i, NET::WMName | NET::WMVisibleName);
