--- src/appfinder-window.c	2021-01-18 12:29:48.000000000 +1300
+++ src/appfinder-window-patch.c	2022-03-04 00:19:38.079994076 +1300
@@ -124,6 +124,7 @@
                                                                        GtkTreeIter                 *a,
                                                                        GtkTreeIter                 *b,
                                                                        gpointer                     data);
+gboolean force_execute;
 
 struct _XfceAppfinderWindowClass
 {
@@ -217,6 +218,7 @@
   GtkCellRenderer    *cell;
   gint                integer;
 
+  force_execute = TRUE;  
   window->channel = xfconf_channel_get ("xfce4-appfinder");
   window->last_window_height = xfconf_channel_get_int (window->channel, "/last/window-height", DEFAULT_WINDOW_HEIGHT);
 
@@ -1384,9 +1386,16 @@
       return TRUE;
     }
 
+  if (force_execute == TRUE && (event->keyval == GDK_KEY_Return || event->keyval == GDK_KEY_KP_Enter))
+  {
+    xfce_appfinder_window_execute (window, TRUE);
+    return FALSE;
+  }
+
   if (event->keyval == GDK_KEY_Up ||
       event->keyval == GDK_KEY_Down)
     {
+      force_execute = FALSE;
       expand = (event->keyval == GDK_KEY_Down);
       is_expanded = gtk_widget_get_visible (window->paned);
 
@@ -1415,6 +1424,10 @@
           gtk_tree_path_free (path);
         }
     }
+   else
+     {
+       force_execute = TRUE;
+     }
 
   return FALSE;
 }
