From 14da2d4a5e7478c720ba4debdce8a965a2415dae Mon Sep 17 00:00:00 2001
From: Theo Linkspfeifer <lastonestanding@tutanota.com>
Date: Thu, 3 Oct 2024 14:07:03 +0200
Subject: [PATCH] Fix use-after-free in thunar_dialogs_show_job_ask_replace
 (#1466)

Fixes: 53690166
---
 thunar/thunar-dialogs.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/thunar/thunar-dialogs.c b/thunar/thunar-dialogs.c
index 871a9f7d..35e4cf8c 100644
--- thunar/thunar-dialogs.c
+++ thunar/thunar-dialogs.c
@@ -881,10 +881,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 
   parent_file = thunar_file_get_parent (dst_file, NULL);
   if (parent_file != NULL)
-    {
-      parent_string = thunar_file_get_basename (parent_file);
-      g_object_unref (parent_file);
-    }
+    parent_string = thunar_file_get_basename (parent_file);
 
   if (thunar_file_is_symlink (dst_file))
     /* TRANSLATORS: First part of replace dialog sentence */
@@ -896,6 +893,9 @@ G_GNUC_END_IGNORE_DEPRECATIONS
     /* TRANSLATORS: First part of replace dialog sentence */
     text = g_strdup_printf (_("Replace the existing file in \"%s\""), parent_string);
 
+  if (parent_file != NULL)
+    g_object_unref (parent_file);
+
   /* next row */
   row++;
 
-- 
2.46.2

