--- src/mobile-helpers.c	2024-01-02 14:29:22.000000000 +0100
+++ src/mobile-helpers.c.patched	2025-11-08 14:11:38.713006000 +0100
@@ -93,15 +93,15 @@
 mobile_helper_get_quality_icon_name (guint32 quality)
 {
 	if (quality > 80)
-		return "nm-signal-100";
+		return "nm-signal-100-symbolic";
 	else if (quality > 55)
-		return "nm-signal-75";
+		return "nm-signal-75-symbolic";
 	else if (quality > 30)
-		return "nm-signal-50";
+		return "nm-signal-50-symbolic";
 	else if (quality > 5)
-		return "nm-signal-25";
+		return "nm-signal-25-symbolic";
 	else
-		return "nm-signal-00";
+		return "nm-signal-00-symbolic";
 }
 
 const char *
