--- meson.build	2025-05-28 12:40:41.519448481 +0000
+++ meson.build.patched	2025-05-28 12:40:50.171448890 +0000
@@ -202,21 +202,6 @@
 # Header checks
 
 stdatomic_dependencies = []
-if not cc.check_header('stdatomic.h')
-    if cc.get_id() == 'msvc'
-        # we have a custom replacement for MSVC
-        stdatomic_dependencies += declare_dependency(
-            include_directories : include_directories('include/compat/msvc'),
-        )
-    elif cc.compiles('''int main() { int v = 0; return __atomic_fetch_add(&v, 1, __ATOMIC_SEQ_CST); }''',
-                     name : 'GCC-style atomics', args : test_args)
-        stdatomic_dependencies += declare_dependency(
-            include_directories : include_directories('include/compat/gcc'),
-        )
-    else
-        error('Atomics not supported')
-    endif
-endif
 
 if host_machine.cpu_family().startswith('wasm')
     # enable atomics + bulk-memory features
