--- etc/xdg/menus/xfce-applications.menu	2022-12-29 12:15:32.000000000 +0000
+++ etc/xdg/menus/xfce-applications.menu.patched	2021-01-18 21:34:19.000000000 +0000
@@ -6,24 +6,16 @@
 
     <DefaultAppDirs/>
     <DefaultDirectoryDirs/>
-    <DefaultMergeDirs/>
 
     <Include>
         <Category>X-Xfce-Toplevel</Category>
     </Include>
 
-    <Layout>
-        <Filename>xfce4-run.desktop</Filename>
-        <Separator/>
-        <Filename>exo-terminal-emulator.desktop</Filename>
-        <Filename>exo-file-manager.desktop</Filename>
-        <Filename>exo-mail-reader.desktop</Filename>
-        <Filename>exo-web-browser.desktop</Filename>
-        <Separator/>
-        <Menuname>Settings</Menuname>
-        <Separator/>
+    <Layout>        
         <Merge type="all"/>
+        <Menuname>Settings</Menuname>
         <Separator/>
+        <Filename>xfce4-run.desktop</Filename>
         <Filename>xfce4-about.desktop</Filename>
         <Filename>xfce4-session-logout.desktop</Filename>
     </Layout>
@@ -159,5 +151,6 @@
             <All/>
         </Include>
     </Menu>
+    <DefaultMergeDirs/>
 
 </Menu>
