--- src/image-view.c	2023-08-18 21:22:39.900933861 +1200
+++ src/image-view-patched.c	2023-08-18 21:23:45.346926658 +1200
@@ -269,7 +269,7 @@
         {
             cairo_rectangle_int_t rectangle;
             cairo_region_get_rectangle(region, i, &rectangle);
-            paint( iv, &rectangle, GDK_INTERP_NEAREST, cr );
+            paint( iv, &rectangle, GDK_INTERP_BILINEAR, cr );
         }
 
         cairo_region_destroy (region);
@@ -306,7 +306,7 @@
         for( i = 0; i < n_rects; ++i )
         {
             // GdkRectangle& rect = rects[i];
-            paint( iv, rects + i, GDK_INTERP_NEAREST );
+            paint( iv, rects + i, GDK_INTERP_BILINEAR );
         }
         g_free( rects );
 
