diff --git a/meson.build b/meson.build
index e5f0a5e3..8fe1659f 100644
--- meson.build
+++ meson.build.patched
@@ -114,16 +114,22 @@ enable_session_selector = get_option('session_selector')
 session_bin_deps += dependency('gio-unix-2.0', version: glib_req_version)
 
 # Check for systemd
+use_systemd = get_option('logind_provider') == 'systemd'
+if use_systemd
+  libsystemd_dep = dependency('libsystemd', version: '>= 209', required: true)
+  session_bin_deps += libsystemd_dep
+elif get_option('logind_provider') == 'elogind'
+  libelogind_dep = dependency('libelogind', version: '>= 209', required: true)
+  session_bin_deps += libelogind_dep
+endif
+
 systemd_userunitdir = get_option('systemduserunitdir')
-if systemd_userunitdir == ''
+if systemd_userunitdir == '' and use_systemd
   systemd_dep = dependency('systemd', version: '>= 242', required: true)
   systemd_userunitdir = systemd_dep.get_variable(pkgconfig: 'systemduserunitdir',
                                                  pkgconfig_define: ['prefix', session_prefix])
 endif
 
-libsystemd_dep = dependency('libsystemd', version: '>= 209', required: true)
-session_bin_deps += libsystemd_dep
-
 configure_file(
   output: 'config.h',
   configuration: config_h