diff --git a/meson_options.txt b/meson_options.txt
index 96c253fd..93023db6 100644
--- meson_options.txt
+++ meson_options.txt.patched
@@ -1,4 +1,5 @@
 option('deprecation_flags', type: 'boolean', value: false, description: 'use *_DISABLE_DEPRECATED flags')
+option('logind_provider', type: 'combo', choices: ['systemd', 'elogind'], value: 'systemd', description: 'logind implementation')
 option('session_selector', type: 'boolean', value: false, description: 'enable building a custom session selector dialog')
 option('systemduserunitdir', type: 'string', description: 'Directory for systemd user service files')
 option('docbook', type: 'boolean', value: true, description: 'build documentation')
-- 
GitLab
