--- gnome-session/gsm-manager.c
+++ gnome-session/gsm-manager.c
@@ -40,9 +40,17 @@
 #include "gsm-manager.h"
 #include "org.gnome.SessionManager.h"
 
+#if defined(HAVE_SYSTEMD) || defined(HAVE_ELOGIND)
 #include <systemd/sd-journal.h>
+#endif
 
+#ifdef HAVE_SYSTEMD
 #include <systemd/sd-daemon.h>
+#else
+/* So we don't need to add ifdef's everywhere */
+#define sd_notify(u, m)            do {} while (0)
+#define sd_notifyf(u, m, ...)      do {} while (0)
+#endif
 
 #include "gsm-store.h"
 #include "gsm-inhibitor.h"
@@ -276,10 +284,12 @@ on_required_app_failure (GsmManager  *manager,
                 allow_logout = !_log_out_is_locked_down (manager);
         }
 
+#ifdef ENABLE_SYSTEMD_JOURNAL
         sd_journal_send ("MESSAGE_ID=%s", GSM_MANAGER_UNRECOVERABLE_FAILURE_MSGID,
                          "PRIORITY=%d", 3,
                          "MESSAGE=Unrecoverable failure in required component %s", app_id,
                          NULL);
+#endif
 
         gsm_fail_whale_dialog_we_failed (FALSE,
                                          allow_logout,
@@ -304,10 +314,12 @@ on_display_server_failure (GsmManager *manager,
                 extensions = NULL;
         }
 
+#ifdef ENABLE_SYSTEMD_JOURNAL
         sd_journal_send ("MESSAGE_ID=%s", GSM_MANAGER_UNRECOVERABLE_FAILURE_MSGID,
                          "PRIORITY=%d", 3,
                          "MESSAGE=Unrecoverable failure in required component %s", app_id,
                          NULL);
+#endif
 
         gsm_quit ();
 }
@@ -962,6 +974,7 @@ _client_stop (const char *id,
         return FALSE;
 }
 
+#if defined(HAVE_SYSTEMD) || defined(HAVE_ELOGIND)
 static void
 maybe_restart_user_bus (GsmManager *manager)
 {
@@ -994,6 +1007,7 @@ maybe_restart_user_bus (GsmManager *manager)
                 g_debug ("GsmManager: reloading user bus failed: %s", error->message);
         }
 }
+#endif
 
 static void
 do_phase_exit (GsmManager *manager)
@@ -1010,9 +1010,10 @@
                                    NULL);
         }
 
+#if defined(HAVE_SYSTEMD) || defined(HAVE_ELOGIND)
         if (!priv->systemd_managed)
                 maybe_restart_user_bus (manager);
-
+#endif
         end_phase (manager);
 }
 
