--- gnome-session/gsm-systemd.c
+++ gnome-session/gsm-systemd.c
@@ -21,6 +21,7 @@
 #include "config.h"
 #include "gsm-systemd.h"
 
+#if defined(HAVE_SYSTEMD) || defined(HAVE_ELOGIND)
 #include <errno.h>
 #include <string.h>
 #include <stdlib.h>
@@ -1169,4 +1170,13 @@ sd_proxy_signal_cb (GDBusProxy  *proxy,
         }
 }
 
+#else
+
+GsmSystemd *
+gsm_systemd_new (void)
+{
+        return NULL;
+}
+
+#endif
 
