--- gnome-session/gsm-util.h
+++ gnome-session/gsm-util.h
@@ -53,6 +53,7 @@ const char * const * gsm_util_listenv               (void);
 const char * const * gsm_util_get_variable_blacklist(void);
 
 gboolean    gsm_util_export_activation_environment  (GError     **error);
+#if defined(HAVE_SYSTEMD) || defined(HAVE_ELOGIND)
 gboolean    gsm_util_export_user_environment        (GError     **error);
 gboolean    gsm_util_systemd_unit_is_active         (const char  *unit,
                                                      GError     **error);
@@ -60,7 +61,7 @@ gboolean    gsm_util_start_systemd_unit             (const char  *unit,
                                                      const char  *mode,
                                                      GError     **error);
 gboolean    gsm_util_systemd_reset_failed           (GError     **error);
-
+#endif
 
 void        gsm_quit                                (void);
 
