--- obconf-qt/src/tree.cpp	2024-05-01 16:47:01.440218286 +0200
+++ obconf-qt/src/tree-obconf-qt.cpp.patched	2024-05-01 16:43:57.845209601 +0200
@@ -16,7 +16,13 @@
    See the COPYING file for a copy of the GNU General Public License.
 */
 
-#include <QX11Info>
+#include <QtGlobal>
+
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+  #include <QApplication>
+#elif
+  #include <QX11Info>
+#endif
 #include <QMessageBox>
 
 #include "tree.h"
@@ -115,12 +121,26 @@
     g_free(p);
 
     if (!err) {
+        #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+          auto x11NativeInterface = qApp->nativeInterface<QNativeInterface::QX11Application>();
+          auto display = x11NativeInterface->display();
+          auto appRootWindow = XDefaultRootWindow(x11NativeInterface->display());
+        #endif
+    
         XEvent ce;
 
         ce.xclient.type = ClientMessage;
-        ce.xclient.display = QX11Info::display();
+        #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+          ce.xclient.display = display;
+        #else
+          ce.xclient.display = QX11Info::display();
+        #endif
         ce.xclient.message_type = XInternAtom(ce.xclient.display, "_OB_CONTROL", false);
-        Window root = QX11Info::appRootWindow();
+        #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+          Window root = appRootWindow;
+        #else
+          Window root = QX11Info::appRootWindow();
+        #endif
         ce.xclient.window = root;
         ce.xclient.format = 32;
         ce.xclient.data.l[0] = 1; /* reconfigure */
