--- obconf-qt/src/windows.cpp	2024-04-24 17:17:14.606518841 +0200
+++ obconf-qt/src/windows.cpp.patched	2024-04-24 19:08:58.778835982 +0200
@@ -20,11 +20,15 @@
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
+#include <QtGlobal>
+
 #include "maindialog.h"
 #include <obrender/render.h>
 #include "tree.h"
 
-#include <QX11Info>
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+  #include <QX11Info>
+#endif
 // FIXME: how to support XCB or Wayland?
 #include <X11/Xlib.h>
 
