--- meson.build	2024-01-04 14:37:37.090971322 +0100
+++ meson.build.patched	2024-01-04 14:38:05.634971049 +0100
@@ -1,7 +1,7 @@
 project('pipewire', ['c' ],
   version : '1.0.0',
   license : [ 'MIT', 'LGPL-2.1-or-later', 'GPL-2.0-only' ],
-  meson_version : '>= 0.61.1',
+  meson_version : '>= 0.59.4',
   default_options : [ 'warning_level=3',
                       'c_std=gnu11',
                       'cpp_std=c++17',
