--- lxqt-powermanagement/CMakeLists.txt	2023-09-04 21:54:15.061947842 +0200
+++ lxqt-powermanagement/CMakeLists.txt.patched	2023-09-04 21:54:19.175947803 +0200
@@ -16,7 +16,7 @@
 option(UPDATE_TRANSLATIONS "Update source translation translations/*.ts files" OFF)
 
 set(LXQTBT_MINIMUM_VERSION "0.13.0")
-set(KF5_MINIMUM_VERSION "5.101.0")
+set(KF5_MINIMUM_VERSION "5.36.0")
 set(LXQT_MINIMUM_VERSION "1.3.0")
 set(QT_MINIMUM_VERSION "5.15.0")
 
