--- lxqt-config/lxqt-config-monitor/loadsettings.cpp	2023-10-14 21:01:26.000000000 +0200
+++ lxqt-config/lxqt-config-monitor/loadsettings.cpp.patched	2023-10-14 21:46:54.485972224 +0200
@@ -133,7 +133,7 @@
                 if( !output->isConnected() )
                     continue;
                 output->setEnabled( monitor.enabled );
-                output->setPriority( monitor.primary ? 1 : 0 );
+                output->setPrimary( monitor.primary );
                 output->setPos( QPoint(monitor.xPos, monitor.yPos) );
                 output->setRotation( (KScreen::Output::Rotation)(monitor.rotation) );
                 // output->setCurrentModeId could fail. KScreen sometimes changes mode Id.
