--- lxqt-panel/plugin-taskbar/lxqttaskgroup.cpp	2023-09-04 22:04:03.000000000 +0200
+++ lxqt-panel/plugin-taskbar/lxqttaskgroup-patched.cpp	2023-09-04 23:05:34.420907031 +0200
@@ -38,7 +38,7 @@
 #include <QStringBuilder>
 #include <QMenu>
 #include <XdgIcon>
-#include <KWindowSystem/KX11Extras>
+#include <KF5/KWindowSystem/KWindowSystem>
 #include <QX11Info>
 #include <functional>
 
@@ -58,8 +58,8 @@
     setText(groupName);
 
     connect(this,                  &LXQtTaskGroup::clicked,               this, &LXQtTaskGroup::onClicked);
-    connect(KX11Extras::self(),    &KX11Extras::currentDesktopChanged,    this, &LXQtTaskGroup::onDesktopChanged);
-    connect(KX11Extras::self(),    &KX11Extras::activeWindowChanged,      this, &LXQtTaskGroup::onActiveWindowChanged);
+    connect(KWindowSystem::self(),    &KWindowSystem::currentDesktopChanged,    this, &LXQtTaskGroup::onDesktopChanged);
+    connect(KWindowSystem::self(),    &KWindowSystem::activeWindowChanged,      this, &LXQtTaskGroup::onActiveWindowChanged);
     connect(parent,                &LXQtTaskBar::buttonRotationRefreshed, this, &LXQtTaskGroup::setAutoRotation);
     connect(parent,                &LXQtTaskBar::refreshIconGeometry,     this, &LXQtTaskGroup::refreshIconsGeometry);
     connect(parent,                &LXQtTaskBar::buttonStyleRefreshed,    this, &LXQtTaskGroup::setToolButtonsStyle);
@@ -97,7 +97,7 @@
 void LXQtTaskGroup::closeGroup()
 {
     for (LXQtTaskButton *button : qAsConst(mButtonHash) )
-        if (button->isOnDesktop(KX11Extras::currentDesktop()))
+        if (button->isOnDesktop(KWindowSystem::currentDesktop()))
             button->closeApplication();
 }
 
@@ -305,7 +305,7 @@
 {
     if (visibleButtonsCount() > 1)
     {
-        setChecked(mButtonHash.contains(KX11Extras::activeWindow()));
+        setChecked(mButtonHash.contains(KWindowSystem::activeWindow()));
         setPopupVisible(true);
     }
 }
@@ -384,7 +384,7 @@
     const int showDesktop = taskbar->showDesktopNum();
     for(LXQtTaskButton * btn : qAsConst(mButtonHash))
     {
-        bool visible = taskbar->isShowOnlyOneDesktopTasks() ? btn->isOnDesktop(0 == showDesktop ? KX11Extras::currentDesktop() : showDesktop) : true;
+        bool visible = taskbar->isShowOnlyOneDesktopTasks() ? btn->isOnDesktop(0 == showDesktop ? KWindowSystem::currentDesktop() : showDesktop) : true;
         visible &= taskbar->isShowOnlyCurrentScreenTasks() ? btn->isOnCurrentScreen() : true;
         visible &= taskbar->isShowOnlyMinimizedTasks() ? btn->isMinimized() : true;
         btn->setVisible(visible);
