--- thunar/thunar-folder.c	2025-10-02 18:09:29.892992929 +0200
+++ thunar/thunar-folder.c.patched	2025-10-02 18:23:20.726032231 +0200
@@ -399,7 +399,10 @@
     g_source_remove (folder->files_update_timeout_source_id);
 
   if (folder->monitor != NULL)
-    g_signal_handlers_disconnect_by_data (folder->monitor, folder);
+    {
+      g_signal_handlers_disconnect_by_data (folder->monitor, folder);
+      g_object_unref (folder->monitor);
+    }
 
   if (folder->corresponding_file)
     {
