--- usr/share/cmake-4.2/Modules/FindLua.cmake	2025-12-23 23:25:33.497656110 +0100
+++ usr/share/cmake-4.2/Modules/FindLua.cmake.patched	2025-12-23 23:25:26.979655802 +0100
@@ -144,7 +144,7 @@
 
 # this is a function only to have all the variables inside go away automatically
 function(_lua_get_versions)
-    set(LUA_VERSIONS5 5.4 5.3 5.2 5.1 5.0)
+    set(LUA_VERSIONS5 5.5 5.4 5.3 5.2 5.1 5.0)
 
     if (Lua_FIND_VERSION_EXACT)
         if (Lua_FIND_VERSION_COUNT GREATER 1)
