--- video/out/opengl/context_drm_egl.c	2023-07-23 19:10:36.000000000 +0200
+++ video/out/opengl/context_drm_egl.c.patched	2023-11-15 23:16:13.567472523 +0100
@@ -497,9 +497,12 @@
 
         eglMakeCurrent(p->egl.display, EGL_NO_SURFACE, EGL_NO_SURFACE,
                        EGL_NO_CONTEXT);
-        eglDestroyContext(p->egl.display, p->egl.context);
-        eglDestroySurface(p->egl.display, p->egl.surface);
-        gbm_surface_destroy(p->gbm.surface);
+        if (p->egl.display) {
+            eglDestroyContext(p->egl.display, p->egl.context);
+            eglDestroySurface(p->egl.display, p->egl.surface);
+        }
+        if (p->gbm.surface)
+            gbm_surface_destroy(p->gbm.surface);
         eglTerminate(p->egl.display);
         gbm_device_destroy(p->gbm.device);
         p->egl.context = EGL_NO_CONTEXT;
