diff --git a/localedata/locales/en_US b/localedata/locales/en_US
--- a/localedata/locales/en_US
+++ b/localedata/locales/en_US
@@ -121,7 +121,7 @@
 % different from d_t_fmt for historical reasons and has been different
 % since 2000 when date_fmt was added as a GNU extension.  At the end
 % of 2018 it was adjusted to use 12H time (bug 24046) instead of 24H.
-date_fmt "%a %b %e %r %Z %Y"
+%date_fmt "%a %d %b %Y %r %Z"
 %
 % Strings for AM/PM
 %
