--- panels/info-overview/cc-info-overview-panel.c	2023-03-17 17:05:05.000000000 +0100
+++ panels/info-overview/cc-info-overview-panel.c.patched	2024-08-29 16:15:45.156200533 +0200
@@ -83,6 +83,24 @@
   char **current;
 } VersionData;
 
+static char* get_gnome_shell_version() {
+   FILE *fp = popen("gnome-shell --version | cut -d \" \" -f 3", "r");
+   if (fp == NULL)
+       return "Not Available";
+
+   char *version = NULL;
+   size_t len = 0;
+   ssize_t read;
+
+   if ((read = getline(&version, &len, fp)) != -1) {
+       if (version[read - 1] == '\n')
+           version[read - 1] = '\0';
+   }
+
+   pclose(fp);
+   return version;
+}
+
 static void
 version_data_free (VersionData *data)
 {
@@ -746,24 +764,6 @@
   return ram_total;
 }
 
-static char *
-get_gnome_version (GDBusProxy *proxy)
-{
-  g_autoptr(GVariant) variant = NULL;
-  const char *gnome_version = NULL;
-  if (!proxy)
-    return NULL;
-
-  variant = g_dbus_proxy_get_cached_property (proxy, "ShellVersion");
-  if (!variant)
-    return NULL;
-
-  gnome_version = g_variant_get_string (variant, NULL);
-  if (!gnome_version || *gnome_version == '\0')
-    return NULL;
-  return g_strdup (gnome_version);
-}
-
 static void
 shell_proxy_ready (GObject             *source,
                    GAsyncResult        *res,
@@ -782,18 +782,9 @@
       g_warning ("Failed to contact gnome-shell: %s", error->message);
     }
 
-  gnome_version = get_gnome_version (proxy);
+  gnome_version = get_gnome_shell_version ();
 
-  if (!gnome_version)
-    {
-      /* translators: this is the placeholder string when the GNOME Shell
-       * version couldn't be loaded, eg. “GNOME Version: Not Available” */
-      cc_list_row_set_secondary_label (self->gnome_version_row, _("Not Available"));
-    }
-  else
-    {
-      cc_list_row_set_secondary_label (self->gnome_version_row, gnome_version);
-    }
+  cc_list_row_set_secondary_label (self->gnome_version_row, gnome_version);
 }
 
 static void
