--- panels/info-overview/cc-info-overview-panel.ui	2023-03-17 17:05:05.000000000 +0100
+++ panels/info-overview/cc-info-overview-panel.ui.patched	2024-08-29 16:18:10.815207424 +0200
@@ -111,7 +111,7 @@
             <child>
               <object class="CcListRow" id="gnome_version_row">
                 <property name="activatable">False</property>
-                <property name="title" translatable="yes">GNOME Version</property>
+                <property name="title" translatable="yes">GNOME Shell</property>
                 <property name="secondary-label">3.0</property>
               </object>
             </child>
