--- lxqt-config-monitor/kscreenutils.cpp	2023-10-14 21:01:26.665998238 +0200
+++ lxqt-config-monitor/kscreenutils.cpp.patched	2023-10-14 21:46:14.346972607 +0200
@@ -63,7 +63,7 @@
         output->setPos(pos);
         output->setEnabled(true);
         //first left one as primary
-        output->setPriority(width == 0 ? 1 : 0);
+        output->setPrimary(width == 0);
         KScreen::ModePtr mode(output->currentMode());
         //if (!mode)
         {
