--- lxqt-config-monitor/monitorwidget.cpp	2023-10-14 21:01:26.000000000 +0200
+++ lxqt-config-monitor/monitorwidget.cpp.patched	2023-10-14 21:47:54.530971651 +0200
@@ -138,7 +138,7 @@
     if (config->connectedOutputs().count() == 1) {
         setOnlyMonitor(true);
         // There isn't always a primary output. Gross.
-        output->setPriority(1);
+        output->setPrimary(true);
     }
 
     ui.xPosSpinBox->setValue(output->pos().x());
@@ -208,7 +208,7 @@
     ui.xPosSpinBox->setVisible(idx == ExtendDisplay);
     ui.yPosSpinBox->setVisible(idx == ExtendDisplay);
 
-    output->setPriority(idx == PrimaryDisplay ? 1 : 0);
+    output->setPrimary(idx == PrimaryDisplay);
     if(idx == PrimaryDisplay)
         emit primaryOutputChanged(this);
 }
