diff --git a/menu/lxqt-applications.menu b/menu/lxqt-applications.menu
--- a/menu/lxqt-applications.menu
+++ b/menu/lxqt-applications.menu
@@ -214,14 +214,6 @@
 		</Include>
 	</Menu>	 <!-- End Leave -->
 
-	<!-- Screensaver -->
-	<Menu>
-		<Name>Screensaver</Name>
-		<Include>
-		<Filename>lxqt-lockscreen.desktop</Filename>
-		</Include>
-	</Menu> <!-- End Screensaver -->
-
 	<Layout>
 		<Merge type="files"/>
 		<Merge type="menus"/>
