--- src/main.rs	2024-08-11 15:02:19.768448300 +0200
+++ src/main.rs.patched	2024-08-11 15:02:10.260928200 +0200
@@ -12,15 +12,26 @@
 use crate::config::VERSION;
 
 fn main() -> anyhow::Result<()> {
-    tracing_subscriber::registry()
-        .with(tracing_journald::layer()?)
-        .with(fmt::layer())
-        .with(
-            EnvFilter::builder()
-                .with_default_directive(LevelFilter::WARN.into())
-                .from_env_lossy(),
-        )
-        .try_init()?;
+    let trace = tracing_subscriber::registry();
+
+    let env_filter = EnvFilter::builder()
+        .with_default_directive(LevelFilter::WARN.into())
+        .from_env_lossy();
+    #[cfg(feature = "systemd")]
+    if let Ok(journald) = tracing_journald::layer() {
+        trace
+            .with(journald)
+            .with(fmt::layer())
+            .with(env_filter)
+            .try_init()?;
+    } else {
+        trace.with(fmt::layer()).with(env_filter).try_init()?;
+        tracing::warn!("Failed to connect to journald")
+    }
+
+    #[cfg(not(feature = "systemd"))]
+    trace.with(fmt::layer()).with(env_filter).try_init()?;
+
     log_panics::init();
 
     info!("cosmic-notifications ({})", APP_ID);
@@ -31,4 +42,4 @@
 
     app::run()?;
     Ok(())
-}
+}
\ No newline at end of file
