--- gdm-40.0.new/data/pam-exherbo/gdm-fingerprint.pam	2021-03-30 10:27:29.859585500 -0300
+++ gdm-40.0/data/pam-exherbo/gdm-fingerprint.pam	2021-07-03 20:02:09.009058407 -0300
@@ -1,6 +1,10 @@
 account  include  system-login
 
 auth     substack fingerprint-auth
+auth     optional pam_env.so
+auth     required pam_shells.so
+auth     required pam_nologin.so
+auth     required pam_permit.so
 auth     optional pam_gnome_keyring.so
 
 password required pam_deny.so
