--- gdm-40.0/data/pam-lfs/gdm-launch-environment.pam	2021-03-30 10:27:29.906585000 -0300
+++ gdm-40.0.new/data/pam-lfs/gdm-launch-environment.pam	2021-07-03 16:42:34.890012949 -0300
@@ -10,6 +10,7 @@
 password required       pam_deny.so
 
 session  required       pam_succeed_if.so audit quiet_success user = gdm
+-session optional       pam_elogind.so
 -session optional       pam_systemd.so
 session  optional       pam_keyinit.so force revoke
 session  optional       pam_permit.so
