--- etc/rc.d/rc.cpufreq	2024-08-08 15:55:11.083007667 +0200
+++ etc/rc.d/rc.cpufreq-patched	2024-08-08 15:55:29.379008532 +0200
@@ -33,7 +33,7 @@
 # provides power savings on Intel processors while avoiding the ramp-up lag
 # present when using the powersave governor (which is the default if ondemand
 # is requested on these machines):
-if [ "$(cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_driver 2> /dev/null)" = "intel_pstate" ]; then
+if [ "$(cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_driver 2> /dev/null)" = "intel_pstate" -o "$(cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_driver 2> /dev/null)" = "amd-pstate-epp" ]; then
   SCALING_GOVERNOR="performance"
 fi
 
