diff --git a/panels/system/about/cc-system-details-window.blp b/panels/system/about/cc-system-details-window.blp
--- a/panels/system/about/cc-system-details-window.blp
+++ b/panels/system/about/cc-system-details-window.blp
@@ -126,7 +126,7 @@
             }
 
             $CcInfoEntry gnome_version_row {
-              label: _("GNOME Version");
+              label: _("GNOME Shell");
             }
 
             $CcInfoEntry windowing_system_row {
