--- panels/system/about/cc-system-details-window.blp	2025-08-20 13:14:44.394537786 +0200
+++ panels/system/about/cc-system-details-window.blp.patched	2025-08-20 13:14:54.152538981 +0200
@@ -126,7 +126,7 @@
             }
 
             $CcInfoEntry gnome_version_row {
-              label: _("GNOME Version");
+              label: _("GNOME Shell");
             }
 
             $CcInfoEntry windowing_system_row {
