--- configure.ac	2024-02-22 01:58:07.000000000 +0100
+++ configure.ac.patched	2024-02-26 13:55:43.705170473 +0100
@@ -99,7 +99,6 @@
 LIBMATE_PANEL_APPLET_REQUIRED=1.17.0
 LIBGTOP_REQUIRED=2.12.0
 LIBCANBERRA_GTK_REQUIRED=0.4
-MATE_DESKTOP_REQUIRED=1.27.1
 UDISKS2_REQUIRED=1.90.0
 
 AC_SUBST(GTK_REQUIRED)
@@ -125,10 +124,6 @@
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
-PKG_CHECK_MODULES(MATE_DESKTOP, mate-desktop-2.0 >= $MATE_DESKTOP_REQUIRED)
-AC_SUBST(MATE_DESKTOP_CFLAGS)
-AC_SUBST(MATE_DESKTOP_LIBS)
-
 AC_ARG_ENABLE([disk-image-mounter],
                [AS_HELP_STRING([--enable-disk-image-mounter=@<:@yes/no@:>@],
                                [Whether to build the Disk Image Mounter utility])],
