--- a/cli/cli.cc	2026-01-27 00:59:25.000000000 +0100
+++ b/cli/cli.cc.patched	2026-01-29 22:29:52.000000000 +0100
@@ -351,8 +367,8 @@
 
     tr_ctorSetPaused(ctor, TR_FORCE, false);
 
-    if (tr_sys_path_exists(torrentPath) ? tr_ctorSetMetainfoFromFile(ctor, torrentPath, nullptr) :
-                                          tr_ctorSetMetainfoFromMagnetLink(ctor, torrentPath, nullptr))
+    if (tr_sys_path_exists(torrentPath) ? tr_ctorSetMetainfoFromFile(ctor, torrentPath) :
+                                          tr_ctorSetMetainfoFromMagnetLink(ctor, torrentPath))
     {
         // all good
     }
