--- a/daemon/daemon.cc	2026-01-27 00:59:25.000000000 +0100
+++ b/daemon/daemon.cc.patched	2026-01-29 22:29:52.000000000 +0100
@@ -223,9 +218,8 @@
         }
         else
         {
-            content.push_back('\0'); // zero-terminated string
-            auto const* data = std::data(content);
-            if (!tr_ctorSetMetainfoFromMagnetLink(ctor, data, nullptr))
+            auto const content_sv = std::string_view{ content.data(), content.size() };
+            if (!tr_ctorSetMetainfoFromMagnetLink(ctor, content_sv))
             {
                 retry = true;
             }
