--- a/libtransmission/transmission.h	2026-01-27 00:59:25.000000000 +0100
+++ b/libtransmission/transmission.h.patched	2026-01-29 22:29:52.000000000 +0100
@@ -773,7 +741,7 @@
 void tr_ctorSetDeleteSource(tr_ctor* ctor, bool delete_source);
 
 /** @brief Set the constructor's metainfo from a magnet link */
-bool tr_ctorSetMetainfoFromMagnetLink(tr_ctor* ctor, char const* magnet, tr_error* error);
+bool tr_ctorSetMetainfoFromMagnetLink(tr_ctor* ctor, std::string_view magnet, tr_error* error = nullptr);
 
 tr_torrent_metainfo const* tr_ctorGetMetainfo(tr_ctor const* ctor);
 
