--- src/icon-grid.c	2022-12-02 20:07:10.862604005 +1300
+++ src/icon-grid-patched.c	2022-12-02 20:21:06.733519109 +1300
@@ -392,7 +392,7 @@
     }
     panel_icon_grid_size_request(widget, &requisition);
     if (minimal_width)
-        *minimal_width = requisition.width;
+		*minimal_width = ig->constrain_width ? 2 : requisition.width;
     if (natural_width)
         *natural_width = requisition.width;
 }
