--- daemon/gdm-manager.c	2025-08-31 20:33:43.217217282 +0200
+++ daemon/gdm-manager.c.patched	2025-08-31 20:35:38.731222747 +0200
@@ -2549,16 +2549,20 @@
         return TRUE;
 }
 
+
 static void
 on_graphics_unsupported (GdmLocalDisplayFactory *factory,
                          GdmManager             *manager)
 {
+#ifdef  WITH_PLYMOUTH
         if (manager->plymouth_is_running) {
                 plymouth_quit_without_transition ();
                 manager->plymouth_is_running = FALSE;
         }
+#endif
 }
 
+
 void
 gdm_manager_stop (GdmManager *manager)
 {
