--- libmate-panel-applet/mate-panel-applet-gsettings.c	2023-10-13 23:11:04.110761864 +0200
+++ libmate-panel-applet/mate-panel-applet-gsettings.patched	2023-10-13 23:11:38.073760991 +0200
@@ -90,6 +90,12 @@
 static void
 register_dconf_editor_relocatable_schema (const gchar *schema, const gchar *path)
 {
+    GSettingsSchema * dconf_editor_schema = g_settings_schema_source_lookup (
+                            g_settings_schema_source_get_default(), 
+                            "ca.desrt.dconf-editor.Settings", FALSE);                                 
+    if (dconf_editor_schema == NULL)
+        return;
+
     GSettings *dconf_editor_settings;
     dconf_editor_settings = g_settings_new ("ca.desrt.dconf-editor.Settings");
 
