--- mate-panel/panel-profile.c	2023-10-08 21:08:13.902740800 +0200
+++ mate-panel/panel-profile-patched.c	2023-10-08 21:07:38.210204400 +0200
@@ -862,6 +862,12 @@
 static void
 unregister_dconf_editor_relocatable_schema (const gchar *path)
 {
+    GSettingsSchema * res = g_settings_schema_source_lookup (
+                            g_settings_schema_source_get_default(), 
+                            "ca.desrt.dconf-editor.Settings", FALSE);                                 
+    if (res == NULL)
+        return;
+
     GSettings *dconf_editor_settings;
     dconf_editor_settings = g_settings_new ("ca.desrt.dconf-editor.Settings");
 
