--- Makefile	2009-11-06 01:57:50.000000000 +0100
+++ Makefile	2013-07-18 10:57:32.482149252 +0200
@@ -1,13 +1,13 @@
 include mkinc.mak
-ifeq ($(OS),Windows_NT)
-EXEEXT=.exe
-endif
-CFLAGS=-O2
-CPPFLAGS=-Iinclude
-all: libstemmer.a stemwords$(EXEEXT)
-libstemmer.a: $(snowball_sources:.c=.o)
-	$(AR) -cru $@ $^
-stemwords$(EXEEXT): examples/stemwords.o libstemmer.a
-	$(CC) $(CFLAGS) -o $@ $^
+CFLAGS+=-Iinclude -fPIC
+all: libstemmer.so libstemmer.a stemwords
+libstemmer.so: $(snowball_sources:.c=.o)
+	$(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname,$@.0d -o $@.0d.0.0 $^
+	ln -s $@.0d.0.0 $@.0d
+	ln -s $@.0d.0.0 $@
+libstemmer.a: $(snowball_sources:.c=.o)
+	$(AR) -crs $@ $^
+stemwords: examples/stemwords.o libstemmer.so
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
 clean:
-	rm -f stemwords$(EXEEXT) libstemmer.a *.o src_c/*.o examples/*.o runtime/*.o libstemmer/*.o
+	rm -f stemwords libstemmer.so* libstemmer.a *.o src_c/*.o runtime/*.o libstemmer/*.o
