--- src/meson.build	2023-12-01 12:34:34.518067075 +0100
+++ src/meson.build.patched	2023-12-01 12:34:28.013066576 +0100
@@ -241,7 +241,7 @@
 lib = library('placebo', sources,
   c_args: ['-DPL_EXPORT'],
   install: true,
-  dependencies: build_deps + glad_dep,
+  dependencies: build_deps + [glad_dep],
   soversion: apiver,
   include_directories: [ inc, vulkan_headers_inc, fast_float_inc ],
   link_args: link_args,
