--- src/meson.build	2023-11-22 14:09:26.350966721 +0100
+++ src/meson.build.patched	2023-11-22 14:09:19.092967520 +0100
@@ -85,7 +85,6 @@
   'utils/libav.h',
   'utils/libav_internal.h',
   'utils/upload.h',
-  'vulkan.h',
 ]
 
 sources = [
@@ -157,7 +156,6 @@
 subdir('glsl')
 subdir('d3d11')
 subdir('opengl')
-subdir('vulkan')
 
 lcms = dependency('lcms2', version: '>=2.9', required: get_option('lcms'))
 components.set('lcms', lcms.found())
@@ -167,14 +165,14 @@
 endif
 
 # Check to see if libplacebo built this way is sane
-if not (components.get('vulkan') or components.get('opengl') or components.get('d3d11'))
+if not (components.get('opengl') or components.get('d3d11'))
   warning('Building without any graphics API. libplacebo built this way still ' +
           'has some limited use (e.g. generating GLSL shaders), but most of ' +
           'its functionality will be missing or impaired!')
 endif
 
 has_spirv = components.get('shaderc') or components.get('glslang')
-needs_spirv = components.get('vulkan') or components.get('d3d11')
+needs_spirv = components.get('d3d11')
 if needs_spirv and not has_spirv
   warning('Building without any GLSL compiler (shaderc, glslang), but with ' +
           'APIs required that require one (vulkan, d3d11). This build is very ' +
@@ -243,7 +241,7 @@
   install: true,
   dependencies: build_deps + glad_dep,
   soversion: apiver,
-  include_directories: [ inc, vulkan_headers_inc, fast_float_inc ],
+  include_directories: [ inc, fast_float_inc ],
   link_args: link_args,
   link_depends: link_depends,
   gnu_symbol_visibility: 'hidden',
