--- src/xfdesktop-file-icon-manager.c.orig	2024-10-14 14:11:52.000000000 +0200
+++ src/xfdesktop-file-icon-manager.c	2024-12-26 03:57:54.822816568 +0100
@@ -1445,7 +1445,7 @@
     if (mdata->position_config == NULL) {
         g_assert(candidates != NULL);
 
-        GtkBuilder *builder = gtk_builder_new_from_resource("/org/xfce/xfdesktop/monitor-candidates-chooser.glade");
+        GtkBuilder *builder = gtk_builder_new_from_resource("/org/xfce/xfdesktop/settings/xfdesktop-monitor-chooser-ui.glade");
         g_assert(builder != NULL);
 
         GtkWidget *dialog = GTK_WIDGET(gtk_builder_get_object(builder, "monitor_candidates_chooser"));
