--- src/core/core.cpp	2023-09-04 22:04:11.000000000 +0200
+++ src/core/core-patched.cpp	2023-09-04 23:51:40.752880650 +0200
@@ -32,8 +32,7 @@
 #include <XdgMimeApps>
 #include <qt5xdg/XdgDesktopFile>
 
-#include <KWindowSystem/KWindowSystem>
-#include <KWindowSystem/KX11Extras>
+#include <KF5/KWindowSystem/KWindowSystem>
 #include <xcb/xfixes.h>
 
 #ifdef X11_XCB_FOUND
@@ -269,15 +268,15 @@
     if (screen == nullptr)
         screen = QGuiApplication::screens().at(0);
 
-    WId wnd = KX11Extras::activeWindow();
+    WId wnd = KWindowSystem::activeWindow();
 
     // this window screenshot will be invalid
     // if there's no active window or the active window is ours
-    bool invalid(!wnd || !KX11Extras::hasWId(wnd) || (_wnd && _wnd->winId() == wnd));
+    bool invalid(!wnd || !KWindowSystem::hasWId(wnd) || (_wnd && _wnd->winId() == wnd));
     if (!invalid)
     { // or if it isn't on the current desktop
         KWindowInfo info(wnd, NET::WMDesktop);
-        invalid = info.valid() && !info.isOnDesktop(KX11Extras::currentDesktop());
+        invalid = info.valid() && !info.isOnDesktop(KWindowSystem::currentDesktop());
         if (!invalid)
         { // or if it is a desktop or panel/dock
             info = KWindowInfo(wnd, NET::WMWindowType);
