--- qps/src/infobar.cpp	2023-05-20 15:31:24.081040884 +1200
+++ qps/src/infobar-patched.cpp	2023-05-20 15:30:55.111044073 +1200
@@ -452,8 +452,7 @@
         height = pf_char_height() + 4;
         x = x_cpu->xpluswidth() + 10;
 #ifdef LINUX
-        used = procview->mem_total - procview->mem_free -
-               procview->mem_buffers - procview->mem_cached;
+        used = procview->mem_total - procview->mem_free;
         width = drawSPECTRUM(p, x, 0, "MEM", procview->mem_total, used,
                              procview->mem_cached, procview->mem_buffers);
 #endif
