--- lxqt-session/lxqt-session/src/procreaper.cpp	2025-02-02 12:10:41.000000000 +0100
+++ lxqt-session/lxqt-session/src/procreaper.cpp.patched	2025-02-02 12:18:36.561335892 +0100
@@ -119,10 +119,10 @@
     pids_stack * stack = nullptr;
     while ((stack = procps_pids_get(info, PIDS_FETCH_TASKS_ONLY)))
     {
-        const int ppid = PIDS_VAL(rel_ppid, s_int, stack, info);
+        const int ppid = PIDS_VAL(rel_ppid, s_int, stack);
         if (ppid == my_pid)
         {
-            const int tgid = PIDS_VAL(rel_tgid, s_int, stack, info);
+            const int tgid = PIDS_VAL(rel_tgid, s_int, stack);
             children.push_back(tgid);
         }
     }
