--- lxqt-panel/plugin-taskbar/lxqttaskbar.cpp	2023-09-04 22:04:03.000000000 +0200
+++ lxqt-panel/plugin-taskbar/lxqttaskbar-patched.cpp	2023-09-04 23:01:26.401909397 +0200
@@ -93,10 +93,10 @@
     connect(mSignalMapper, &QSignalMapper::mappedInt, this, &LXQtTaskBar::activateTask);
     QTimer::singleShot(0, this, &LXQtTaskBar::registerShortcuts);
 
-    connect(KX11Extras::self(), static_cast<void (KX11Extras::*)(WId, NET::Properties, NET::Properties2)>(&KX11Extras::windowChanged)
+    connect(KWindowSystem::self(), static_cast<void (KWindowSystem::*)(WId, NET::Properties, NET::Properties2)>(&KWindowSystem::windowChanged)
             , this, &LXQtTaskBar::onWindowChanged);
-    connect(KX11Extras::self(), &KX11Extras::windowAdded, this, &LXQtTaskBar::onWindowAdded);
-    connect(KX11Extras::self(), &KX11Extras::windowRemoved, this, &LXQtTaskBar::onWindowRemoved);
+    connect(KWindowSystem::self(), &KWindowSystem::windowAdded, this, &LXQtTaskBar::onWindowAdded);
+    connect(KWindowSystem::self(), &KWindowSystem::windowRemoved, this, &LXQtTaskBar::onWindowRemoved);
 }
 
 /************************************************
@@ -356,7 +356,7 @@
 {
     QList<WId> new_list;
     // Just add new windows to groups, deleting is up to the groups
-    const auto wnds = KX11Extras::stackingOrder();
+    const auto wnds = KWindowSystem::stackingOrder();
     for (auto const wnd: wnds)
     {
         if (acceptWindow(wnd))
