--- lxqt-panel/panel/plugin.cpp	2023-09-04 22:04:03.000000000 +0200
+++ lxqt-panel/panel/plugin-patched.cpp	2023-09-04 23:06:34.062906463 +0200
@@ -31,8 +31,7 @@
 #include "pluginsettings_p.h"
 #include "lxqtpanel.h"
 
-#include <KWindowSystem/KX11Extras>
-
+#include <KWindowSystem>
 #include <QDebug>
 #include <QProcessEnvironment>
 #include <QStringList>
@@ -523,8 +522,8 @@
     mConfigDialog->activateWindow();
 
     WId wid = mConfigDialog->windowHandle()->winId();
-    KX11Extras::activateWindow(wid);
-    KX11Extras::setOnDesktop(wid, KX11Extras::currentDesktop());
+    KWindowSystem::activateWindow(wid);
+    KWindowSystem::setOnDesktop(wid, KWindowSystem::currentDesktop());
 }
 
 
