diff --git a/doc/sample-xpdfrc b/doc/sample-xpdfrc
--- a/doc/sample-xpdfrc
+++ b/doc/sample-xpdfrc
@@ -29,20 +29,20 @@
 # installed in a "standard" location, xpdf will find them
 # automatically.)
 
-#fontFile Times-Roman		/usr/local/share/ghostscript/fonts/n021003l.pfb
-#fontFile Times-Italic		/usr/local/share/ghostscript/fonts/n021023l.pfb
-#fontFile Times-Bold		/usr/local/share/ghostscript/fonts/n021004l.pfb
-#fontFile Times-BoldItalic	/usr/local/share/ghostscript/fonts/n021024l.pfb
-#fontFile Helvetica		/usr/local/share/ghostscript/fonts/n019003l.pfb
-#fontFile Helvetica-Oblique	/usr/local/share/ghostscript/fonts/n019023l.pfb
-#fontFile Helvetica-Bold		/usr/local/share/ghostscript/fonts/n019004l.pfb
-#fontFile Helvetica-BoldOblique	/usr/local/share/ghostscript/fonts/n019024l.pfb
-#fontFile Courier		/usr/local/share/ghostscript/fonts/n022003l.pfb
-#fontFile Courier-Oblique	/usr/local/share/ghostscript/fonts/n022023l.pfb
-#fontFile Courier-Bold		/usr/local/share/ghostscript/fonts/n022004l.pfb
-#fontFile Courier-BoldOblique	/usr/local/share/ghostscript/fonts/n022024l.pfb
-#fontFile Symbol			/usr/local/share/ghostscript/fonts/s050000l.pfb
-#fontFile ZapfDingbats		/usr/local/share/ghostscript/fonts/d050000l.pfb
+fontFile Times-Roman		/usr/share/fonts/Type1/n021003l.pfb
+fontFile Times-Italic		/usr/share/fonts/Type1/n021023l.pfb
+fontFile Times-Bold		/usr/share/fonts/Type1/n021004l.pfb
+fontFile Times-BoldItalic	/usr/share/fonts/Type1/n021024l.pfb
+fontFile Helvetica		/usr/share/fonts/Type1/n019003l.pfb
+fontFile Helvetica-Oblique	/usr/share/fonts/Type1/n019023l.pfb
+fontFile Helvetica-Bold		/usr/share/fonts/Type1/n019004l.pfb
+fontFile Helvetica-BoldOblique	/usr/share/fonts/Type1/n019024l.pfb
+fontFile Courier		/usr/share/fonts/Type1/n022003l.pfb
+fontFile Courier-Oblique	/usr/share/fonts/Type1/n022023l.pfb
+fontFile Courier-Bold		/usr/share/fonts/Type1/n022004l.pfb
+fontFile Courier-BoldOblique	/usr/share/fonts/Type1/n022024l.pfb
+fontFile Symbol			/usr/share/fonts/Type1/s050000l.pfb
+fontFile ZapfDingbats		/usr/share/fonts/Type1/d050000l.pfb
 
 # If you need to display PDF files that refer to non-embedded fonts,
 # you should add one or more fontDir options to point to the
