--- Cargo.toml	2024-08-11 15:22:02.896504300 +0200
+++ Cargo.toml.patched	2024-08-11 15:23:13.531686200 +0200
@@ -39,11 +39,7 @@
 ] }
 tokio-util = "0.7"
 tracing = "0.1"
-tracing-journald = "0.3"
+tracing-journald = { version = "0.3", optional = true }
 tracing-subscriber = { version = "0.3", features = ["env-filter"] }
 zbus = { version = "4.3.0", default-features = false, features = ["tokio"] }
 cosmic-notifications-util = { git = "https://github.com/pop-os/cosmic-notifications" }
-
-[features]
-systemd = ["dep:zbus_systemd"]
-default = ["systemd"]
