--- etc/xdg/menus/xfce-applications.menu	2022-12-30 09:55:51.000000000 +0000
+++ etc/xdg/menus/xfce-applications.menu.patched	2021-01-18 21:34:19.000000000 +0000
@@ -11,18 +11,11 @@
         <Category>X-Xfce-Toplevel</Category>
     </Include>
 
-    <Layout>
-        <Filename>xfce4-run.desktop</Filename>
-        <Separator/>
-        <Filename>xfce4-terminal-emulator.desktop</Filename>
-        <Filename>xfce4-file-manager.desktop</Filename>
-        <Filename>xfce4-mail-reader.desktop</Filename>
-        <Filename>xfce4-web-browser.desktop</Filename>
-        <Separator/>
-        <Menuname>Settings</Menuname>
-        <Separator/>
+    <Layout>        
         <Merge type="all"/>
+        <Menuname>Settings</Menuname>
         <Separator/>
+        <Filename>xfce4-run.desktop</Filename>
         <Filename>xfce4-about.desktop</Filename>
         <Filename>xfce4-session-logout.desktop</Filename>
     </Layout>
